### PR TITLE
Minor thread sampling tweaks

### PR DESCRIPTION
**Pull request by johnbley**
- document buffer format a bit
- clean up some warnings
- reduce size of length prefix for strings since they are limited to 1k anyways
- fix type issue when NameCache was reused for class names

### CODE REVIEW

File: tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp [hunk 1]
```diff
@@ -152,8 +177,8 @@ void ThreadSamplesBuffer::StartBatch()
 void ThreadSamplesBuffer::StartSample(ThreadID id, ThreadState* state, ThreadSpanContext context)
 {
     CHECK_SAMPLES_BUFFER_LENGTH();
-    writeByte(0x02);
-    writeInt(id);
+    writeByte(THREAD_SAMPLES_START_SAMPLE);
+    writeInt((int)id); // FIXME not really sure how to map this to anything; needs more research
```
Comment (Kielek) [inline tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp:181]: Could you please create separate jira item to fix it?

File: tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp [hunk 2]
```diff
@@ -618,7 +648,7 @@ WSTRING* NameCache::get(FunctionID key)
     return found->second->second;
 }
 
-void NameCache::put(FunctionID key, WSTRING* val)
+void NameCache::put(UINT_PTR key, WSTRING* val)
 {
     auto pair = std::pair<FunctionID, WSTRING*>(key, val);
     list.push_front(pair);
```
Comment (Kielek) [inline tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp:654]: any chance to fix warning also in line 657 - if (map.size() > maxSize)

**Review by Kielek — APPROVED** [review]

**Review by pellared — APPROVED** [review]
wouldn't changing the type of `NameCache.maxSize` to `size_t` fix some warnings?

File: tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp (hunk 2 above)
Comment (pellared) [reply tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp:654]: wouldn't changing the type of `NameCache.maxSize` to `size_t` fix some warnings?

File: tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp (hunk 1 above)
Comment (johnbley) [reply tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp:181]: (done)

File: tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp (hunk 2 above)
Comment (johnbley) [reply tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp:654]: Good idea, done.

File: tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp [hunk 3]
```diff
@@ -75,7 +75,7 @@ int ThreadSampling_ConsumeOneThreadSample(int len, unsigned char* buf)
     {
         return 0;
     }
-    int toUseLen = min(toUse->size(), len);
+    int toUseLen = (int) min(toUse->size(), len);
```
Comment (pellared) [inline tracer/src/Datadog.Trace.ClrProfiler.Native/ThreadSampler.cpp:78]: @johnbley I guess that `len` argument of `ThreadSampling_ConsumeOneThreadSample` should also be of type `size_t`

**Pull request merged**